Samples/PowerShell/Interface.Cleanup.ps1

```diff
@@ -0,0 +1,31 @@
+# Based on a conversation at the Seattle SWUG about using the API
+# to clean up non-up interfaces, this has been written/provided.
+
+Import-Module SwisPowerShell
+
+$OrionServer = 'localhost'
+$Username = 'admin'
+$Password = ''
+
+$swis = Connect-Swis -Hostname $OrionServer -Username $Username -Password $Password
+
+# query to get node name, interface name, and interface URI of each non-up
+# interface
+$query = "
+    SELECT
+        Nodes.Caption AS NodeName
+        ,Interfaces.Caption AS InterfaceName
+        ,Interfaces.URI
+    FROM Orion.Nodes AS Nodes
+    JOIN Orion.NPM.Interfaces AS Interfaces ON Nodes.NodeID = Interfaces.NodeID
+    WHERE Interfaces.Status != 1
+"
```

Comment:
"Learn something new every day" tip: Use navigation properties. Joins in SWIS are evil.

```
SELECT 
   i.Node.Caption as NodeName,
   i.Caption as InterfaceName,
   i.URI
 FROM Orion.NPM.Interfaces i
 WHERE i.Status != 1
```

is equivalent.

Reply:
Thanks for the update. :)